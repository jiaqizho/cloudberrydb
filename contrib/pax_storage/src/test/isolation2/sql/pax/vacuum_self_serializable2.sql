-- @Description Ensures that a vacuum with serializable works ok
-- 
DROP TABLE IF EXISTS pax_tbl;
DROP TABLE IF EXISTS pax_tbl2;
CREATE TABLE pax_tbl (a INT, b INT);
CREATE TABLE pax_tbl2 (a INT);
INSERT INTO pax_tbl SELECT i as a, i as b FROM generate_series(1, 100) AS i;

DELETE FROM pax_tbl WHERE a <= 30;
1: SET TRANSACTION ISOLATION LEVEL SERIALIZABLE;
1: SELECT COUNT(*) FROM pax_tbl;
1: BEGIN;
1: SELECT COUNT(*) FROM pax_tbl2;
2: SELECT ptblockname, pttupcount FROM get_pax_aux_table_all('pax_tbl');
2: VACUUM pax_tbl;
2: SELECT ptblockname, pttupcount FROM get_pax_aux_table_all('pax_tbl');
1: SELECT COUNT(*) FROM pax_tbl;
1: COMMIT;
