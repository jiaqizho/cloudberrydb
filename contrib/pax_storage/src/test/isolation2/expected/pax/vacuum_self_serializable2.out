-- @Description Ensures that a vacuum with serializable works ok
--
DROP TABLE IF EXISTS pax_tbl;
DROP
DROP TABLE IF EXISTS pax_tbl2;
DROP
CREATE TABLE pax_tbl (a INT, b INT);
CREATE
CREATE TABLE pax_tbl2 (a INT);
CREATE
INSERT INTO pax_tbl SELECT i as a, i as b FROM generate_series(1, 100) AS i;
INSERT 100

DELETE FROM pax_tbl WHERE a <= 30;
DELETE 30
1: SET TRANSACTION ISOLATION LEVEL SERIALIZABLE;
SET
1: SELECT COUNT(*) FROM pax_tbl;
 count 
-------
 70    
(1 row)
1: BEGIN;
BEGIN
1: SELECT COUNT(*) FROM pax_tbl2;
 count 
-------
 0     
(1 row)
2: SELECT ptblockname, pttupcount FROM get_pax_aux_table_all('pax_tbl');
 ptblockname | pttupcount 
-------------+------------
 0           | 25         
 0           | 38         
 0           | 37         
(3 rows)
2: VACUUM pax_tbl;
VACUUM
2: SELECT ptblockname, pttupcount FROM get_pax_aux_table_all('pax_tbl');
 ptblockname | pttupcount 
-------------+------------
 0           | 38         
 0           | 25         
 0           | 37         
(3 rows)
1: SELECT COUNT(*) FROM pax_tbl;
 count 
-------
 70    
(1 row)
1: COMMIT;
COMMIT
